packages/cspell/src/lint.ts

```diff
@@ -23,29 +24,29 @@ export interface FileResult {
     elapsedTimeMs: number;
 }
 
-export interface RunResult {
-    files: number;
-    filesWithIssues: Set<string>;
-    issues: number;
-    errors: number;
-}
+export async function runLint(cfg: CSpellApplicationConfiguration): Promise<RunResult> {
+    if (cfg.options.root) {
+        process.env[cspell.ENV_CSPELL_GLOB_ROOT] = cfg.root;
+    }
+    const configInfo: ConfigInfo = await readConfig(cfg.configFile, cfg.root);
+
+    const reporter = mergeReporters(cfg.reporter, ...loadReporters(configInfo.config));
 
-export function runLint(cfg: CSpellApplicationConfiguration): Promise<RunResult> {
     const configErrors = new Set<string>();
 
     const log: Logger['log'] = (...params) => {
         const msg = format(...params);
-        cfg.emitters.info(msg, 'Info');
+        reporter.info(msg, 'Info');
```

Comment:
This requires a LOT of refactoring because reporter gets passed to cspell-lib via `setLogger` where all calls should be awaited.

I dont think that is really needed - nothing really bad will happen if reporter calls are not awaited

Maybe we should throw Promises away here and allow emitter functions to return `any`?

Reply:
I am fine with not returning promises for now. I think it complicates the `cspell-lib`. Making them `(...) => void` seems to make the most sense. `any` implies that the return value is used.

I would like to support `async` reporters. This can be done by having all the emitters be sync and adding a single
async `close` method to `CSpellReporter`. 


```ts
export interface CSpellReporter {
// ...
close: () => Promise<void>;
}
```

That way a reporter can keep track of all its own async calls and resolve them at once during the close.

There are plenty of `await`s in the code to make sure a reporter's promises get time to run.

Reply:
`close` seems nice

Reply:
actually, result emitter can act like `close` that you suggest
I'll make only ResultEmitter async

Reply:
The ResultEmitter is fine.